packages/core/src/v3/serverOnly/httpServer.ts

```
@@ -141,25 +187,28 @@ export class HttpServer {
+        );
+
+        if (error) {
+          logger.error("Route handler error", { error });
           return reply.empty(500);
         }
       } catch (error) {
         logger.error("Failed to handle request", { error });
         return reply.empty(500);
+      } finally {
+        this.collectMetrics(req, res, startTime);
+        return;
```

Biome (1.9.4)

[error] Unsafe usage of 'return'.

'return' in 'finally' overwrites the control flow statements inside 'try' and 'catch'.

(lint/correctness/noUnsafeFinally)